subsys/console/tty.c

```diff
@@ -91,12 +91,16 @@ int tty_putchar(struct tty_serial *tty, u8_t c)
 	return 0;
 }
 
-u8_t tty_getchar(struct tty_serial *tty)
+int tty_getchar(struct tty_serial *tty)
 {
```

Comment:
don't you check against NULL pointer?

Reply:
> don't you check against NULL pointer?

I don't, just the same as I don't test for a pointer with value of 1, 2, 3, and unlimited (well, asymptotically 2^32 on 32-bit machine) number of other invalid pointers. That's my position when writing small systems like Zephyr. We can discuss requirements re: that separately, as it's implementation detail, not part of API (well, largely: as long as there's ability to communicate back errors - and that should be part of the API, more errors can be added later as needed).